Client/Frontend/Browser/BrowserLongPressHandler.swift

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+
+    ///Check whether long pressed is on valid link. If found valid link then find the rect for link.
+    func evaluateTouchLocation(touchLocation: CGPoint,gestureRecognizer: UILongPressGestureRecognizer) {
+
+        self.webView!.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler: { (elementsJSON: AnyObject!, error: NSError!) -> Void in
+
+            if elementsJSON == nil || (elementsJSON as? String) == nil {
+                return
+            }
+
+            let elementsStr = elementsJSON as String
+            let json = JSON(string: elementsStr)
+            let tags = json["tags"].asArray
+
+            if tags == nil || tags?.count == 0 {
+                return
+            }
+
+            let selectedTags = tags!.filter({ ($0.asString == "A" || $0.asString == "IMG") })
+            if (selectedTags.count == 0) {
+                return
+            }
+            //
+            let hrefElement = json["hrefElement"].asDictionary
+            var elements:[ElementType] = []
+            if hrefElement != nil {
+                let hrefStr: String = hrefElement!["hrefLink"]!.asString!
+                let hrefLinkURL = NSURL(string: hrefStr)
+                let linkRect = self.rectLocationInWebView(self.webView!, locationRect:CGRectFromString(hrefElement!["location"]!.asString!))
+                if self.isValidURL(hrefLinkURL) {
+                    elements.append(ElementType.hrefLink(url: hrefLinkURL!, location: linkRect))
+                }
+            }
+
+            let imageElement = json["imageElement"].asDictionary
+            if imageElement != nil {
+                let imageSrcStr: String = imageElement!["imageSrc"]!.asString!
+                let imageSrc = NSURL(string: imageSrcStr)
+                let linkRect = self.rectLocationInWebView(self.webView!, locationRect:CGRectFromString(imageElement!["location"]!.asString!))
+                if self.isValidURL(imageSrc) {
+                    elements.append(ElementType.ImageSrc(url: imageSrc!, location: linkRect))
+                }
+            }
+
+            var disableGestures:[UIGestureRecognizer] = []
+            self.recursiveBlockOnViewAndSubviews(self.webView!) { view in
+                if let gestureRecognizers = view.gestureRecognizers as? [UIGestureRecognizer] {
+                    for g in gestureRecognizers {
+                        if g != gestureRecognizer  && g.enabled == true {
+                            g.enabled = false
+                            disableGestures.append(g)
+                        }
+                    }
+                }
+            }
+
+            //Calls delegate method
+            self.longPressHandlerDelegate?.webView(self.webView!, elementsDidLongPressed: elements)
+            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(0.5 * Double(NSEC_PER_SEC))), dispatch_get_main_queue()) {
```

Comment:
This looks like an ugly hack. We should definitely look for an alternative approach here.